src/agent/onefuzz/src/az_copy.rs

```diff
@@ -145,16 +183,24 @@ async fn retry_az_impl(mode: Mode, src: &OsStr, dst: &OsStr, args: &[&str]) -> R
                 failure_count + 1
             )
         });
-        match result {
-            Ok(()) => Ok(()),
-            Err(err) => {
-                if !should_always_retry(&err) {
-                    failure_count = failure_counter.fetch_add(1, Ordering::SeqCst);
-                }
-                if failure_count >= RETRY_COUNT {
-                    Err(backoff::Error::Permanent(err))
-                } else {
-                    Err(backoff::Error::Transient(err))
+        let post_copy = fs::metadata(src).await;
+
+        if file_change_detected(src, pre_copy, post_copy) {
+            Err(backoff::Error::Transient(anyhow::Error::msg(
```

Comment:
Why shouldn't these failures subject to our retry limiting policies (the retry counter + `should_always_retry`)?